modules/openapi-generator/src/main/resources/python/api_client.mustache

```diff
@@ -713,7 +731,7 @@ class ApiClient:
                 'Authentication token must be in `query` or `header`'
             )
 
-    def __deserialize_file(self, response):
+    def __deserialize_file(self, response: rest.RESTResponse):
```

Comment:
I think you should pass the "body" parameter from the `deserialize` function here too, and use it instead of `response.data` below?

Reply:
I think a file should not be decoded, but written as it is received from the server. I changed this to write the binary response, before it was opening the file with "wb" mode, but writing the decoded response I believe.

Deserialization should maybe be split into `_deserialize` and `_write_file` methods to make this clearer, but maybe in a later PR?

Reply:
I think "file" responses are still decoded if `_preload_content=True`, which is probably a separate bug. Decoding might fail on some response bodies.

Reply:
If needed, it's probably worth adding more dedicated tests on this specific feature: the parsing of the response and how the files are loaded afterward is a bit intricate and it would best to make sure there are no regressions introduced.

Reply:
"file" responses are an OpenAPI v2 feature (https://swagger.io/docs/specification/2-0/describing-responses/#response-that-returns-a-file). OpenAPI v3 only has "bytearrays": https://swagger.io/docs/specification/describing-responses/#response-that-returns-a-file

I cannot find any OpenAPI v2 example schema to test this with, but the openapi-generator Github page states that OpenAPI v2 is still supported.